### PR TITLE
Update for main page of documentation

### PR DESCRIPTION
**Pull request by IgorWilbert**
## Description

For #1007.

## Roadmap

- [X]  Add project badges;

- [x] Make required syntax changes in README.md (which will actually become README.rst);

- [x] Import content from README to documentation site; 

- [x] Fix image display inside documentation;

- [ ]  Give a more formal title for the links to the API components (better than `horovod.tensorflow`, `horovod.tensorflow.keras` etc.);

### CODE REVIEW

**Comment by IgorWilbert** [comment]
@alsrgv I would like to know your thoughts about it. Thanks!

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+Internally at Uber we found the MPI model to be much more straightforward and require far less code changes than the
+Distributed TensorFlow with parameter servers. See the [Usage](#usage) section for more details.
+
+In addition to being easy to use, Horovod is fast. Below is a chart representing the benchmark that was done on 128
+servers with 4 Pascal GPUs each connected by RoCE-capable 25 Gbit/s network:
+
+.. image:: https://user-images.githubusercontent.com/16640218/38965607-bf5c46ca-4332-11e8-895a-b9c137e86013.png
+   :alt: 512-GPU Benchmark
+
+Horovod achieves 90% scaling efficiency for both Inception V3 and ResNet-101, and 68% scaling efficiency for VGG-16.
```
Comment (alsrgv) [inline README.rst]: Fix the link?

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+Horovod core principles are based on `MPI <http://mpi-forum.org/>`_ concepts such as *size*, *rank*,
+*local rank*, *allreduce*, *allgather* and *broadcast*. See `this page <docs/concepts.md>`_ for more details.
+
+
+Usage
+-----
+
+To use Horovod, make the following additions to your program:
+
+1. Run ``hvd.init()``.
```
Comment (alsrgv) [inline README.rst]: I like the formatting in https://github.com/IgorWilbert/horovod/tree/documentation-readme#usage, can you make it more consistent?  E.g. (1) should have the same font as the first sentence of (2).  (3), (4) shouldn't have sentences abruptly change the formatting.

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+2. To run on 4 machines with 4 GPUs each:
+
+.. code-block:: bash
+
+    horovodrun -np 16 -H server1:4,server2:4,server3:4,server4:4 python train.py
+
+
+3. To run in Docker, see the `Horovod in Docker <docs/docker.md>`_ page.
+
+4. To run in Kubernetes, see `Kubeflow <https://github.com/kubeflow/kubeflow/tree/master/kubeflow/mpi-job>`_,
```
Comment (alsrgv) [inline README.rst]: For some reason (4) has different formatting than other sections.

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+
+.. inclusion-marker-start-do-not-remove
+
+.. contents::
+
+
+Horovod is a distributed training framework for TensorFlow, Keras, PyTorch, and MXNet. The goal of Horovod is to make
+distributed Deep Learning fast and easy to use.
+
+.. image:: https://github.com/LFDLFoundation/artwork/raw/master/lfdl/horizontal/color/lfdl-horizontal-color.png
```
Comment (alsrgv) [inline README.rst]: Can we make this logo smaller?  On my preview, it takes up the whole width.

**Comment by alsrgv** [comment]
@sblotner, could you take a look as well?  You can see rendering here - https://github.com/IgorWilbert/horovod/tree/documentation-readme

File: docs/index.rst [hunk 5]
```diff
@@ -4,6 +4,8 @@ Welcome to Horovod's documentation!
 .. toctree::
    :maxdepth: 2
 
+   readme_include
```
Comment (alsrgv) [inline docs/index.rst]: I think the first page needs to have a distillation of the info from our readme, with links to find out more information on the sub-pages.  Bloated README needs to go.

File: README.rst [hunk 6]
```diff
@@ -0,0 +1,415 @@
...
+.. image:: https://pepy.tech/badge/horovod
+   :target: https://pepy.tech/project/horovod
+   :alt: Downloads
+
+.. inclusion-marker-start-do-not-remove
+
+.. contents::
+
+
+Horovod is a distributed training framework for TensorFlow, Keras, PyTorch, and MXNet. The goal of Horovod is to make
```
Comment (sblotner) [inline README.rst]: Recommend moving this above "Contents" so people can see it sooner

File: docs/index.rst (hunk 5 above)
Comment (sblotner) [reply docs/index.rst]: I agree. Maybe the Why not traditional Distributed TensorFlow?, Keras, Estimator API, MXNet, PyTorch, mpi4py sections can be similar to the Inference section, with 1-2 sentences and then link to a sub-page

File: README.rst [hunk 7]
```diff
@@ -0,0 +1,415 @@
...
+about who's involved and how Horovod plays a role, read the LF DL `announcement <https://lfdl.io/press/2018/12/13/lf-deep-learning-welcomes-horovod-distributed-training-framework-as-newest-project/>`_.
+
+
+Why not traditional Distributed TensorFlow?
+-------------------------------------------
+
+The primary motivation for this project is to make it easy to take a single-GPU TensorFlow program and successfully train
+it on many GPUs faster. This has two aspects:
+
+1. How much modification does one have to make to a program to make it distributed, and how easy is it to run it.
```
Comment (sblotner) [inline README.rst]: Change . to ?

File: README.rst [hunk 8]
```diff
@@ -0,0 +1,415 @@
...
+
+
+Install
+------------
+
+To install Horovod:
+
+1. Install `Open MPI <https://www.open-mpi.org/>`_ or another MPI implementation.
+
+Steps to install Open MPI are listed `on this page <https://www.open-mpi.org/faq/?category=building#easy-build>`_.
```
Comment (sblotner) [inline README.rst]: Learn how to install Open MPI on this page

Comment (sblotner) [reply README.rst]: Also, recommend making this a second sentence after "another MPI implementation" (same line)

File: README.rst [hunk 9]
```diff
@@ -0,0 +1,415 @@
...
+
+**Note**: Open MPI 3.1.3 has an issue that may cause hangs.  It is recommended
```
Comment (sblotner) [inline README.rst]: It is recommended to--> The recommended fix is to

File: README.rst [hunk 10]
```diff
@@ -0,0 +1,415 @@
...
+This basic installation is good for laptops and for getting to know Horovod.
+If you're installing Horovod on a server with GPUs, read the `Horovod on GPU <docs/gpus.md>`_ page.
+If you want to use Docker, read the `Horovod in Docker <docs/docker.md>`_ page.
+
+
+Concepts
+------------
+
+Horovod core principles are based on `MPI <http://mpi-forum.org/>`_ concepts such as *size*, *rank*,
+*local rank*, *allreduce*, *allgather* and *broadcast*. See `this page <docs/concepts.md>`_ for more details.
```
Comment (sblotner) [inline README.rst]: add comma after "and"

File: README.rst [hunk 11]
```diff
@@ -0,0 +1,415 @@
...
+
+Keras
+-----
+Horovod supports Keras and regular TensorFlow in similar ways.
+
+See full training `simple <examples/keras_mnist.py>`_ and `advanced <examples/keras_mnist_advanced.py>`_ examples.
+
+**Note**: Keras 2.0.9 has a `known issue <https://github.com/fchollet/keras/issues/8353>`_ that makes each worker allocate
+all GPUs on the server, instead of the GPU assigned by the *local rank*. If you have multiple GPUs per server, upgrade
+to Keras 2.1.2, or downgrade to Keras 2.0.8.
```
Comment (sblotner) [inline README.rst]: remove comma after 2.1.2

File: README.rst [hunk 12]
```diff
@@ -0,0 +1,415 @@
...
+            label = batch.label[0].as_in_context(context)
+            with autograd.record():
+                output = model(data.astype(dtype, copy=False))
+                loss = loss_fn(output, label)
+            loss.backward()
+            trainer.step(batch_size)
+
+
+
+**Note**: There is a `known issue <https://github.com/horovod/horovod/issues/884>`_ when running Horovod with MXNet on a Linux system with GCC version 5.X and above. We recommend users to build MXNet from source following this `guide <https://mxnet.incubator.apache.org/install/build_from_source.html>`_ as a workaround for now.
```
Comment (sblotner) [inline README.rst]: recommend users to build --> recommend that you build

File: README.rst [hunk 13]
```diff
@@ -0,0 +1,415 @@
...
+
+
+PyTorch
+-------
+Horovod supports PyTorch and TensorFlow in similar ways.
+
+Example (also see a full training `example <examples/pytorch_mnist.py>`_)):
```
Comment (sblotner) [inline README.rst]: remove extra ")"

File: README.rst [hunk 14]
```diff
@@ -0,0 +1,415 @@
...
+
+.. image:: https://user-images.githubusercontent.com/16640218/29735271-9e148da0-89ac-11e7-9ae0-11d7a099ac89.png
+   :alt: Horovod Timeline
+
+See `here <docs/timeline.md>`_ for full details and usage instructions.
+
+
+Guides
+------
+1. Run distributed training in Microsoft Azure using `Batch AI and Horovod <https://github.com/Azure/BatchAI/tree/master/recipes/Horovod>`_.
```
Comment (sblotner) [inline README.rst]: Remove the "1." for now

When there is more than one link, it can be an unordered list

File: README.rst [hunk 15]
```diff
@@ -0,0 +1,415 @@
...
+
+Troubleshooting
+---------------
+See the `Troubleshooting <docs/troubleshooting.md>`_ page and please submit the `ticket <https://github.com/uber/horovod/issues/new>`_
```
Comment (sblotner) [inline README.rst]: the ticket --> a ticket

File: README.rst (hunk 1 above)
Comment (IgorWilbert) [reply README.rst]: Fixed, thanks!

File: README.rst (hunk 2 above)
Comment (IgorWilbert) [reply README.rst]: This was not intended. I had some trouble with `code` and **bold** conflics, so I left it out for not. If you really want it, I'll check how to solve it ;)

File: README.rst (hunk 3 above)
Comment (IgorWilbert) [reply README.rst]: Solved, thanks!

File: README.rst (hunk 4 above)
Comment (IgorWilbert) [reply README.rst]: Solved, thanks!

File: README.rst (hunk 6 above)
Comment (IgorWilbert) [reply README.rst]: Good idea, done!

File: docs/index.rst (hunk 5 above)
Comment (IgorWilbert) [reply docs/index.rst]: Sounds good! I'll implement it as soon as we agree about which parts should be subpages

**Comment by IgorWilbert** [comment]
No problem @alsrgv and @sblotner , thanks for the reviews! I made the requested changes. Also, it would be great if you could help me eliminate these console errors. They happen because I use the same keywords for more than one link.
error

**Comment by alsrgv** [comment]
@IgorWilbert, thanks for the updates!  Apparently, you've got to [use double underscores](https://github.com/sphinx-doc/sphinx/issues/3921#issuecomment-315581557) to make Sphinx happy with duplicate references.

Can you rebase your branch with `--signoff`, to make DCO happy?

File: docs/index.rst (hunk 5 above)
Comment (alsrgv) [reply docs/index.rst]: Let's start with Stephanie's suggestion.

**Comment by IgorWilbert** [comment]
@alsrgv all done, thanks! Is there anything I need to do to make the buildkite pass? It seems to fail in `Build Docs`

File: README.rst [hunk 16]
```diff
@@ -0,0 +1,419 @@
...
+-------------------------------------------
+
+The primary motivation for this project is to make it easy to take a single-GPU TensorFlow program and successfully train
+it on many GPUs faster. This has two aspects:
+
+1. How much modification does one have to make to a program to make it distributed, and how easy is it to run it?
+2. How much faster would it run in distributed mode?
+
+Internally at Uber we found the MPI model to be much more straightforward and require far less code changes than the
+Distributed TensorFlow with parameter servers. See the [Usage](#usage) section for more details.
```
Comment (alsrgv) [inline README.rst]: Fix `[Usage](#usage)` link

File: README.rst [hunk 17]
```diff
@@ -0,0 +1,419 @@
...
+   restored from a checkpoint. Alternatively, if you're not using ``MonitoredTrainingSession``, you can simply execute
+   the ``hvd.broadcast_global_variables`` op after global variables have been initialized.
+
+6. Modify your code to save checkpoints only on worker 0 to prevent other workers from corrupting them.
+   This can be accomplished by passing ``checkpoint_dir=None`` to ``tf.train.MonitoredTrainingSession`` if
+   ``hvd.rank() != 0``.
+
+Example (see the `examples <examples/>`_ directory for full training examples):
+
+.. code-block:: bash
```
Comment (alsrgv) [inline README.rst]: Can you change all Python code blocks to be marked up as Python?

File: README.rst [hunk 18]
```diff
@@ -0,0 +1,419 @@
...
+
+.. image:: https://user-images.githubusercontent.com/16640218/29735271-9e148da0-89ac-11e7-9ae0-11d7a099ac89.png
+   :alt: Horovod Timeline
+
+See `here <docs/timeline.md>`__ for full details and usage instructions.
+
+
+Guides
+------
+Run distributed training in Microsoft Azure using `Batch AI and Horovod <https://github.com/Azure/BatchAI/tree/master/recipes/Horovod>`_.
```
Comment (alsrgv) [inline README.rst]: Suggest to keep it with explicit number `1` to highlight that other guides are coming.  We can also add something like "Please submit us links to your user guides to be published here" (@sblotner, what would be a good wording?)

File: README.rst [hunk 19]
```diff
@@ -0,0 +1,419 @@
...
+      Author = {Alexander Sergeev and Mike Del Balso},
+      Journal = {arXiv preprint arXiv:1802.05799},
+      Title = {Horovod: fast and easy distributed deep learning in {TensorFlow}},
+      Year = {2018}
+    }
+
+
+Publications
+------------
+1. Sergeev, A., Del Balso, M. (2017) *Meet Horovod: Uber’s Open Source Distributed Deep Learning Framework for TensorFlow*.
```
Comment (alsrgv) [inline README.rst]: This list doesn't render correctly.

File: README.rst (hunk 18 above)
Comment (sblotner) [reply README.rst]: Sure, maybe "Send us links to any user guides you want to publish on this site"

File: README.rst (hunk 19 above)
Comment (sblotner) [reply README.rst]: May be caused by the line breaks between numbers

File: README.rst (hunk 16 above)
Comment (IgorWilbert) [reply README.rst]: Done, thanks!

File: README.rst (hunk 19 above)
Comment (IgorWilbert) [reply README.rst]: For some reason, I had to add `Retrieved from`, but now it works

File: README.rst (hunk 18 above)
Comment (IgorWilbert) [reply README.rst]: Added, good catch!

File: README.rst (hunk 17 above)
Comment (IgorWilbert) [reply README.rst]: I think `.. code-block:: python` is what you wanted. Let me know if I'm mistaken

**Comment by IgorWilbert** [comment]
Hey! Now I guess the only thing missing is the "distillation of the info from our readme". Do you want me to do it on this PR? If so, I'll start working on it.

**Comment by alsrgv** [comment]
Let's land this change and start a new one. Thanks!!

**Pull request merged**